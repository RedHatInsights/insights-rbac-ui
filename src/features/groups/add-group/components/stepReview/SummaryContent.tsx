import React from 'react';
import { Grid } from '@patternfly/react-core';
import { GridItem } from '@patternfly/react-core';
import { Stack } from '@patternfly/react-core';
import { StackItem } from '@patternfly/react-core';
import { Text } from '@patternfly/react-core/dist/dynamic/components/Text';
import { TextVariants } from '@patternfly/react-core/dist/dynamic/components/Text';
import useFormApi from '@data-driven-forms/react-form-renderer/use-form-api';
import { useIntl } from 'react-intl';
import messages from '../../../../../Messages';
import useChrome from '@redhat-cloud-services/frontend-components/useChrome';
import { useFlag } from '@unleash/proxy-client-react';

interface SummaryContentProps {
  name?: string;
  // Data-driven-forms props
  [key: string]: any;
}

export const SummaryContent: React.FC<SummaryContentProps> = () => {
  const intl = useIntl();
  const formOptions = useFormApi();
  const { isBeta } = useChrome();
  const {
    'group-name': name,
    'group-description': description,
    'users-list': selectedUsers,
    'roles-list': selectedRoles,
    'service-accounts-list': selectedServiceAccounts,
  } = formOptions.getState().values || {};
  const enableServiceAccounts =
    (isBeta() && useFlag('platform.rbac.group-service-accounts')) || (!isBeta() && useFlag('platform.rbac.group-service-accounts.stable'));

  return (
    <div className="rbac">
      <Stack hasGutter>
        <StackItem>
          <Stack hasGutter>
            <StackItem className="rbac-l-stack__item-summary">
              <Grid>
                <GridItem md={3}>
                  <Text component={TextVariants.h4} className="rbac-bold-text">
                    {intl.formatMessage(messages.groupName)}
                  </Text>
                </GridItem>
                <GridItem md={9}>
                  <Text component={TextVariants.p}>{name}</Text>
                </GridItem>
              </Grid>
              <Grid>
                <GridItem md={3}>
                  <Text component={TextVariants.h4} className="rbac-bold-text">
                    {intl.formatMessage(messages.description)}
                  </Text>
                </GridItem>
                <GridItem md={9}>
                  <Text component={TextVariants.p}>{description || intl.formatMessage(messages.none)}</Text>
                </GridItem>
              </Grid>
            </StackItem>
            <StackItem className="rbac-l-stack__item-summary">
              <Grid>
                <GridItem md={3}>
                  <Text component={TextVariants.h4} className="rbac-bold-text">
                    {intl.formatMessage(messages.roles)}
                  </Text>
                </GridItem>
                <GridItem md={9}>
                  <Stack>
                    {selectedRoles && selectedRoles.length > 0 ? (
                      selectedRoles.map((role: any) => (
                        <StackItem key={role.uuid}>
                          <Text component={TextVariants.p}>{role.display_name || role.name}</Text>
                        </StackItem>
                      ))
                    ) : (
                      <StackItem>
                        <Text component={TextVariants.p}>{'No roles selected'}</Text>
                      </StackItem>
                    )}
                  </Stack>
                </GridItem>
              </Grid>
            </StackItem>
            <StackItem className="rbac-l-stack__item-summary">
              <Grid>
                <GridItem md={3}>
                  <Text component={TextVariants.h4} className="rbac-bold-text">
                    {intl.formatMessage(messages.members)}
                  </Text>
                </GridItem>
                <GridItem md={9}>
                  <Stack>
                    {selectedUsers && selectedUsers.length > 0 ? (
                      selectedUsers.map((user: any) => (
                        <StackItem key={user.uuid || user.username}>
                          <Text component={TextVariants.p}>{user.label || user.username}</Text>
                        </StackItem>
                      ))
                    ) : (
                      <StackItem>
                        <Text component={TextVariants.p}>{'No members selected'}</Text>
                      </StackItem>
                    )}
                  </Stack>
                </GridItem>
              </Grid>
            </StackItem>
            {enableServiceAccounts && (
              <StackItem className="rbac-l-stack__item-summary">
                <Grid>
                  <GridItem md={3}>
                    <Text component={TextVariants.h4} className="rbac-bold-text">
                      Service accounts
                    </Text>
                  </GridItem>
                  <GridItem md={9}>
                    <Stack>
                      {selectedServiceAccounts && selectedServiceAccounts.length > 0 ? (
                        selectedServiceAccounts.map((sa: any) => (
                          <StackItem key={sa.uuid || sa.clientId}>
                            <Text component={TextVariants.p}>{sa.name || sa.clientId}</Text>
                          </StackItem>
                        ))
                      ) : (
                        <StackItem>
                          <Text component={TextVariants.p}>{'No service accounts selected'}</Text>
                        </StackItem>
                      )}
                    </Stack>
                  </GridItem>
                </Grid>
              </StackItem>
            )}
          </Stack>
        </StackItem>
      </Stack>
    </div>
  );
};
