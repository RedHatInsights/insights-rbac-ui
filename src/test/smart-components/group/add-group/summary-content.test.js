import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import SummaryContent from '../../../../smart-components/group/add-group/summary-content';

describe('<SummaryContent />', () => {
  let initialProps;

  beforeEach(() => {
    initialProps = {
      formData: {
        values: {
          name: 'TestGroupName',
          description: 'This is a test',
          policy: {
            name: 'TestPolicyName',
            description: 'Test Policy Description',
          },
        },
        selectedUsers: [],
        selectedRoles: [],
      },
    };
  });

  it('should render correctly', () => {
    const wrapper = mount(<SummaryContent {...initialProps} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
