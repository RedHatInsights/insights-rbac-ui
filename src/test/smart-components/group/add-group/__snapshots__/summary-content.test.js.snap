// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SummaryContent /> should render correctly 1`] = `
<SummaryContent
  formData={
    Object {
      "selectedRoles": Array [],
      "selectedUsers": Array [],
      "values": Object {
        "description": "This is a test",
        "name": "TestGroupName",
        "policy": Object {
          "description": "Test Policy Description",
          "name": "TestPolicyName",
        },
      },
    }
  }
>
  <Stack
    hasGutter={true}
  >
    <div
      className="pf-l-stack pf-m-gutter"
    >
      <StackItem>
        <div
          className="pf-l-stack__item"
        >
          <Stack
            hasGutter={true}
          >
            <div
              className="pf-l-stack pf-m-gutter"
            >
              <StackItem>
                <div
                  className="pf-l-stack__item"
                >
                  <TextContent>
                    <div
                      className="pf-c-content"
                    >
                      <Title
                        headingLevel="h4"
                        size="xl"
                      >
                        <h4
                          className="pf-c-title pf-m-xl"
                        >
                          Review details
                        </h4>
                      </Title>
                    </div>
                  </TextContent>
                </div>
              </StackItem>
              <StackItem
                className="ins-c-rbac__summary"
              >
                <div
                  className="pf-l-stack__item ins-c-rbac__summary"
                >
                  <Grid
                    hasGutter={true}
                  >
                    <div
                      className="pf-l-grid pf-m-gutter"
                    >
                      <GridItem
                        span={3}
                      >
                        <div
                          className="pf-l-grid__item pf-m-3-col"
                        >
                          <Text
                            className="pf-c-title"
                            component="h6"
                          >
                            <h6
                              className="pf-c-title"
                              data-pf-content={true}
                            >
                              Group name
                            </h6>
                          </Text>
                        </div>
                      </GridItem>
                      <GridItem
                        span={9}
                      >
                        <div
                          className="pf-l-grid__item pf-m-9-col"
                        >
                          <Text
                            component="p"
                          >
                            <p
                              className=""
                              data-pf-content={true}
                            />
                          </Text>
                        </div>
                      </GridItem>
                    </div>
                  </Grid>
                  <Grid
                    hasGutter={true}
                  >
                    <div
                      className="pf-l-grid pf-m-gutter"
                    >
                      <GridItem
                        span={3}
                      >
                        <div
                          className="pf-l-grid__item pf-m-3-col"
                        >
                          <Text
                            className="pf-c-title"
                            component="h6"
                          >
                            <h6
                              className="pf-c-title"
                              data-pf-content={true}
                            >
                              Group description
                            </h6>
                          </Text>
                        </div>
                      </GridItem>
                      <GridItem
                        span={9}
                      >
                        <div
                          className="pf-l-grid__item pf-m-9-col"
                        >
                          <Text
                            component="p"
                          >
                            <p
                              className=""
                              data-pf-content={true}
                            />
                          </Text>
                        </div>
                      </GridItem>
                    </div>
                  </Grid>
                  <Grid
                    hasGutter={true}
                  >
                    <div
                      className="pf-l-grid pf-m-gutter"
                    >
                      <GridItem
                        span={3}
                      >
                        <div
                          className="pf-l-grid__item pf-m-3-col"
                        >
                          <Text
                            className="pf-c-title"
                            component="h6"
                          >
                            <h6
                              className="pf-c-title"
                              data-pf-content={true}
                            >
                              Roles
                            </h6>
                          </Text>
                        </div>
                      </GridItem>
                      <GridItem
                        span={9}
                      >
                        <div
                          className="pf-l-grid__item pf-m-9-col"
                        >
                          <Text
                            component="p"
                          >
                            <p
                              className=""
                              data-pf-content={true}
                            />
                          </Text>
                        </div>
                      </GridItem>
                    </div>
                  </Grid>
                  <Grid
                    hasGutter={true}
                  >
                    <div
                      className="pf-l-grid pf-m-gutter"
                    >
                      <GridItem
                        span={3}
                      >
                        <div
                          className="pf-l-grid__item pf-m-3-col"
                        >
                          <Text
                            className="pf-c-title"
                            component="h6"
                          >
                            <h6
                              className="pf-c-title"
                              data-pf-content={true}
                            >
                              Members
                            </h6>
                          </Text>
                        </div>
                      </GridItem>
                      <GridItem
                        span={9}
                      >
                        <div
                          className="pf-l-grid__item pf-m-9-col"
                        >
                          <Text
                            component="p"
                          >
                            <p
                              className=""
                              data-pf-content={true}
                            />
                          </Text>
                        </div>
                      </GridItem>
                    </div>
                  </Grid>
                </div>
              </StackItem>
            </div>
          </Stack>
        </div>
      </StackItem>
    </div>
  </Stack>
</SummaryContent>
`;
